Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -81,6 +89,7 @@
   private final List<String> myPriorityFiles = new ArrayList<String>();
 
   @NotNull private final Queue<CompletionInfo> myCompletionInfos = new LinkedList<CompletionInfo>();
+  @NotNull private final Map<String, List<NavigationRegion>> myNavigationData = new THashMap<String, List<NavigationRegion>>();
```

Comment:
Or just create map as Collections.synchronizedMap(new THashMap<String, List<NavigationRegion>>()) - it's the same as adding 2 synchronized blocks.